data/planet.lua

```diff
@@ -759,37 +854,60 @@ local space_connections = {
+    SpaceConnection.length = SpaceConnection.length / ScaleFactor -- Divide each length by the Scale Factor.
+    
+    if math.ceil(SpaceConnection.length) ~= SpaceConnection.length then
+        SpaceConnection.length = math.ceil(SpaceConnection.length) -- Round up to the nearest integer.
+    end
+    
+    if SpaceConnection.length < 100 then
+        SpaceConnection.length = 100 -- Set the length to 100 if the length is less than 100. In my testing, anything less than 100 breaks at speeds reached in the base game.
+    end
+    
+    space_connections[i] = SpaceConnection -- Set the new space connection to the modified one.
 end
 
 data:extend(space_connections)
 
-local planets = {
-    { "mercury", "水星", { "planet-discovery-venus", "space-platform-thruster" } },
-    { "venus", "金星", { "planet-discovery-luna", "space-platform-thruster" } },
-    { "earth", "地球", { "space-platform-thruster" } },
-    { "luna", "月球", { "planet-discovery-earth", "space-platform-thruster" } },
-    { "mars", "火星", { "planet-discovery-luna", "space-platform-thruster" } },
-    { "asteroid-belt-inner", "内小行星带", { "planet-discovery-mars", "space-platform-thruster" } },
-    { "jupiter", "木星", { "planet-discovery-asteroid-belt-inner", "space-platform-thruster" } },
-    { "saturn", "土星", { "planet-discovery-jupiter", "space-platform-thruster" } },
-    { "uranus", "天王星", { "planet-discovery-saturn", "space-platform-thruster" } },
-    { "neptune", "海王星", { "planet-discovery-uranus", "space-platform-thruster" } },
-    { "pluto", "冥王星", { "planet-discovery-neptune", "space-platform-thruster" } },
-    { "asteroid-belt-outer", "柯伊伯带", { "planet-discovery-pluto", "space-platform-thruster" } }
+local PlanetTechnologies = {
+    { "mercury", "水星", { "planet-discovery-venus", "space-platform-thruster" }, false },
+    { "venus", "金星", { "planet-discovery-luna", "space-platform-thruster" }, false },
+    { "earth", "地球", { "space-platform-thruster" }, false },
+    { "luna", "月球", { "planet-discovery-earth", "space-platform-thruster" }, false },
+    { "mars", "火星", { "planet-discovery-luna", "space-platform-thruster" }, false },
+    { "phobos", "火卫一", { "planet-discovery-mars", "space-platform-thruster" }, true },
+    { "deimos", "火卫二", { "planet-discovery-mars", "space-platform-thruster" }, true },
+    { "asteroid-belt-inner", "内小行星带", { "planet-discovery-mars", "space-platform-thruster" }, false },
+    { "jupiter", "木星", { "planet-discovery-asteroid-belt-inner", "space-platform-thruster" }, false },
+    { "saturn", "土星", { "planet-discovery-jupiter", "space-platform-thruster" }, false },
+    { "uranus", "天王星", { "planet-discovery-saturn", "space-platform-thruster" }, false },
+    { "neptune", "海王星", { "planet-discovery-uranus", "space-platform-thruster" }, false },
+    { "pluto", "冥王星", { "planet-discovery-neptune", "space-platform-thruster" }, false },
+    { "asteroid-belt-outer", "柯伊伯带", { "planet-discovery-pluto", "space-platform-thruster" }, false }
```

Comment:
I thought we needed access to the original planet data (containing the prototypes) but we didn't and this is the result. The Boolean values represent if there are icons for these technologies.